server.js

```diff
@@ -16,9 +25,104 @@ const app = express();
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
+// API documentation
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  try {
+    const endpoints = expressListEndpoints(app).map((endpoint) => {
+      if (endpoint.path === "/thoughts") {
+        endpoint.query = {
+          description: `Here you can send a GET request to get the last 20 thoughts in a decending order. As well as posting a new thought to the API with a POST request. Please include a JSON with the message like that: { "message": "My Happy Thought" }`,
+        };
+      }
+      if (endpoint.path === "/thoughts/:thoughtId/like") {
+        endpoint.query = {
+          description:
+            "On this endpoint you can like a thought by sending a POST request. Please provide the Id of the desired thought for it to work.",
+        };
+      }
+      return endpoint;
+    });
+    res.status(200).json(endpoints);
+  } catch (error) {
+    console.error("The following error occured:", error);
+    res
+      .status(500)
+      .send(
+        "Sorry, this page is not available at the moment. Please try again later."
+      );
+  }
+});
+
+//GET the thoughts endpoint
+//Also sorting them in a decending order and limit the thoughts to 20 shown
+app.get("/thoughts", async (req, res) => {
+  try {
+    const allThoughts = await Thought.find()
+      .sort({ createdAt: -1 })
+      .limit(20)
+      .exec();
+
+    if (allThoughts.length > 0) {
+      res.status(200).json(allThoughts);
+    } else {
+      res.status(404).send("No Thoughts were found.");
+    }
+  } catch (error) {
+    console.error("The following error occured:", error);
+    res
+      .status(500)
+      .send(
+        "Sorry, this page is not available at the moment. Please try again later."
+      );
+  }
+});
+
+//POST the thoughts endpoint
+app.post("/thoughts", async (req, res) => {
+  const { message, hearts, createdAt, _id } = req.body;
+
+  try {
+    const thought = await new Thought({
+      message,
+      hearts,
+      createdAt,
+      _id,
+    }).save();
+    res.status(201).json(thought);
```

Comment:
<img width="978" alt="Screenshot 2024-05-20 at 12 21 39" src="https://github.com/Technigo/project-happy-thoughts-api/assets/117687564/c3f4f8bc-3547-4747-afd8-e232cd7154bc">
Great that you tried to use some mongoose method in the Schema to prevent data from being manipulated. But I think "setDefaultsOnInsert" method only works when you update the data by using "update() or findOneAndUpdate()". I also tested it out and noticed it will not assign the hearts to default 0 when I provided the hearts number (see the screenshot).
An easiest way to fix this is to extract the info you want, in this case only the message, and ignore those you don't want.